src/app/services/character.service.ts

```diff
@@ -15,7 +16,7 @@ interface ApiResponse {
 @Injectable({
   providedIn: 'root'
 })
-export class CharacterService {
+export class CharacterService implements CharacterServiceAbstract {
```

Comment:
Why the use of abstract class?

Reply:
so our mock class has the same functionality for testing

Reply:
we added an abstract class so that our mocked class will be strict and have to implement the functions inside of the abstract class. as well as any other classes that implements it.

Reply:
Could also have used an interface or `extends` to create a subclass of the mock, which is common. For example:

```
class MyMock extends MyComponenetClass {}
```

Reply:
is the current implementation acceptable as is?